### PR TITLE
misc improvements to pointcloud viz:

### PR DESCRIPTION
**Pull request by tocttou**
1. added proper offset handling
2. fields are now read according to their field.datatype
3. added intensity setting
4. optimized rendering and required memory

### CODE REVIEW

File: src/utils/constants.js [hunk 1]
```diff
@@ -5,7 +5,7 @@ export const OBJECT_TYPE_ARROW = 'Arrow';
 export const OBJECT_TYPE_AXES = 'Axes';
 export const OBJECT_TYPE_FLAT_ARROW = 'FlatArrow';
 
-export const MAX_POINTCLOUD_POINTS = 100000;
+export const MAX_POINTCLOUD_POINTS = 500000;
```
Comment (tocttou) [inline src/utils/constants.js]: Many pointclouds do not fit inside 100,000 points. 500,000 seems to be a better default to avoid confusion to the users (comes at the cost of extra memory). Ideally the buffer should resize if the number of data points change.

File: src/utils/constants.js [hunk 2]
```diff
@@ -279,6 +295,9 @@ export const DEFAULT_OPTIONS_PLANNINGSCENE = {};
 
 export const DEFAULT_OPTIONS_POINTCLOUD = {
   compression: 'cbor',
+  colorChannel: 'rgb',
```
Comment (chaitanya-deep) [inline src/utils/constants.js]: Replace with `POINTCLOUD_COLOR_CHANNELS.RGB`

File: src/viz/PointCloud.js [hunk 3]
```diff
@@ -2,81 +2,132 @@ import * as THREE from 'three';
...
 class PointCloud extends Core {
   constructor(ros, topicName, options = DEFAULT_OPTIONS_POINTCLOUD) {
-    super(ros, topicName, MESSAGE_TYPE_POINTCLOUD2, options);
+    super(ros, topicName, MESSAGE_TYPE_POINTCLOUD2, {
+      ...DEFAULT_OPTIONS_POINTCLOUD,
+      ...options,
+    });
     const cloudMaterial = new THREE.PointsMaterial({
-      size: 0.1,
+      size: 0.05,
```
Comment (chaitanya-deep) [inline src/viz/PointCloud.js]: Shouldn't this be set from options?

File: src/viz/PointCloud.js [hunk 4]
```diff
@@ -85,29 +136,31 @@ class PointCloud extends Core {
...
-    const { geometry } = this.object;
+  updatePointCloudGeometry(positions, colors, normals) {
+    const { geometry, material } = this.object;
+    if (
+      material.size !== this.options.size &&
+      !Number.isNaN(this.options.size)
+    ) {
+      material.size = this.options.size;
+      material.needsUpdate = true;
+    } else if (material.needsUpdate) {
```
Comment (chaitanya-deep) [inline src/viz/PointCloud.js]: Why is this required? Won't this introduce unwanted bugs when something else other than change in size requires the material to update?

Comment (tocttou) [reply src/viz/PointCloud.js]: Right. I kept it there to turn off re-renders on subsequent cycles, but I think this call can be safely moved to `this.updateOptions` instead to avoid the bugs.
